tokio/src/executor/task/harness.rs

```diff
@@ -51,7 +55,7 @@ where
     /// Panics raised while polling the future are handled.
     ///
     /// Returns `true` if the task needs to be scheduled again
-    pub(super) fn poll(mut self, executor: &mut dyn FnMut() -> Option<NonNull<S>>) -> bool {
+    pub(super) fn poll(mut self, executor: &mut dyn FnMut() -> Option<NonNull<()>>) -> bool {
```

Comment:
I think this signature is now a little misleading. It'd be useful to at least say in the docs that the expectation is that it is safe to cast the `NonNull<()>` into `NonNull<S>`. Arguably this `fn` should be marked as `unsafe`.

Reply:
I added a safety comment.